keps/sig-node/3063-dynamic-resource-allocation/kubelet.puml

```diff
@@ -0,0 +1,12 @@
+@startuml
+!theme reddress-lightblue
+
+(*) --> "Pod scheduled to the node"
+-->[ **Kubelet** pick up the pod ] "Pod references resource claim"
+-->[ **Kubelet** calls NodePrepareResource(ResourceClaimUID, AllocationParameters)\l **Resource Plugin** creates CDI JSON file(s) ] "CDI JSON files created"
+-->[ **Resource Plugin** returns device Ids ] "Resource prepared"
+-->[ **Kubelet** passes device Ids to the CRI runtime\l **CRI runtime** uses CDI JSON file(s) to update OCI container config\l **CRI runtime** creates pod containers ] "All Pod containers created"
+-->[ ... ] "Pod runs"
+-->[ ... ] "Pod terminated"
+-->[ **Kubelet** calls NodeUnprepareResource\l **Resource plugin** removes CDI JSON file(s)\l **Kubelet** removes pod from ResourceClaim.ReservedFor ] (*)
```

Comment:
Error handling is not covered here (which is okay, it is a simplification) but also not explained in the main text. This might be worth adding.